src/main/java/fi/aalto/cs/apluscourses/utils/parser/FeedbackParser.java

```diff
@@ -0,0 +1,57 @@
+package fi.aalto.cs.apluscourses.utils.parser;
+
+import java.util.regex.Pattern;
+import org.jetbrains.annotations.NotNull;
+import org.jsoup.Jsoup;
+
+public class FeedbackParser {
+  private FeedbackParser() {
+  }
+
+  /**
+   * Parses TestResults for O1 from an HTML string.
+   */
+  public static TestResults parseO1TestResults(@NotNull String htmlString) {
```

Comment:
I suggest we make each of these a separate subclass/implementation of `FeedbackParser` such as `O1FeedbackParser`. Then course config file could define which feedback parser it uses like `"feedbackParser": "O1"`.